sdcm/nemesis.py

```diff
@@ -1023,82 +1047,75 @@ def _choose_file_for_destroy(self, ks_cfs):
-                                                                                     ks_cf_for_destroy=ks_cf_for_destroy))
-                self.log.debug('Error: {}'.format(error))
+            if not (file_for_destroy := self.replace_full_file_name_to_prefix(one_file, ks_cf_for_destroy)):
                 continue
 
-            file_for_destroy = one_file.replace(file_name, file_name_template + '-*')
             self.log.debug('Selected files for destroy: {}'.format(file_for_destroy))
             if file_for_destroy:
-                break
+                if return_one_file:
+                    break
+                all_files.append(file_for_destroy)
 
         if not file_for_destroy:
             raise NoFilesFoundToDestroy('Data file for destroy is not found in {}'.format(ks_cf_for_destroy))
 
-        return file_for_destroy
+        return file_for_destroy if return_one_file else all_files
 
-    @decorate_with_context(ignore_ycsb_connection_refused)
-    def _destroy_data_and_restart_scylla(self):
-        def find_five_files_to_destroy():
-            all_files_to_destroy = []
-            try:
-                for _ in range(5):
-                    for _ in range(3):
-                        file_for_destroy = self._choose_file_for_destroy(tables)
-                        if file_for_destroy not in all_files_to_destroy:
-                            all_files_to_destroy.append(file_for_destroy)
-                            break
-            except NoFilesFoundToDestroy:
-                if not all_files_to_destroy:
-                    raise
-            return all_files_to_destroy
-        ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node, filter_empty_tables=False)
-        if not ks_cfs:
-            raise UnsupportedNemesis(
-                'Non-system keyspace and table are not found. CorruptThenRepair nemesis can\'t be run')
+    def get_all_sstables(self, tables: list[str], node: BaseNode = None):
+        """
+        :param tables: list of tables. Format of name: <keyspace_name.table_name>
+        :param node: get SStables from the node
+        """
+        node = node or self.target_node
 
-        tables = []
-        errors = []
-        with self.cluster.cql_connection_patient(self.target_node) as session:
-            for table in ks_cfs:
-                has_data, error = self.cluster.is_table_has_data(session=session, table_name=table)
-                if has_data:
-                    tables.append(table)
-                if error:
-                    errors.append(error)
-                if len(tables) > 20:
-                    break
+        sstables = []
+        for ks_cf in tables:
+            sstable_util = SstableUtils(db_node=node, ks_cf=ks_cf)
+            ks_cf_sstables = sstable_util.get_sstables()
+            sstables.extend(ks_cf_sstables)
+            self.log.info("Sstables for % ks_cf are: %s", ks_cf, ks_cf_sstables)
```

Comment:
these sstable lists prints to log is duplicated - here and in line below.
It may also be better to print as a debug message and not info.

Reply:
removed